src/main/kotlin/org/rust/ide/annotator/RsAttrErrorAnnotator.kt

```diff
@@ -22,13 +22,42 @@ class RsAttrErrorAnnotator : AnnotatorBase() {
                 checkMetaBadDelim(element, holder)
                 checkAttrTemplateCompatible(element, holder)
                 checkLiteralSuffix(element, holder)
+
+                when (element.name) {
+                    "deprecated" -> checkDeprecatedAttr(element, holder)
+                }
             }
 
             is RsDocAndAttributeOwner -> checkRsDocAndAttributeOwner(element, holder)
         }
     }
 }
 
+private fun checkDeprecatedAttr(element: RsMetaItem, holder: AnnotationHolder) {
+    if (element.templateType != AttributeTemplateType.List) return
+    val usedArgs = mutableSetOf<String>()
+    for (arg in element.metaItemArgsList) {
+        if (arg !is RsMetaItem) continue
+        val argName = arg.name ?: continue
+        if (arg.litExpr == null) {
+            RsDiagnostic.IncorrectItemInDeprecatedAttr(arg).addToHolder(holder)
+        }
+        when (argName) {
+            // "suggestion" argument is used only in compiler development
+            "since", "note", "suggestion" -> {
```

Comment:
I suppose this check should be moved inside `"since", "note", "suggestion" -> {` branch since rustc shows only `error[E0551]: incorrect meta item` in this case: `#[deprecated(foobarbranch)]`.
Also, I think this error should not be shown in the case of `#[deprecated(since = )]` since it's a kind o f syntax error, rustc shows "expected unsuffixed literal or identifier, found `<eof>`" in this case

```suggestion
        when (argName) {
            if (arg.eq == null) {
                RsDiagnostic.IncorrectItemInDeprecatedAttr(arg).addToHolder(holder)
            }
            // "suggestion" argument is used only in compiler development
            "since", "note", "suggestion" -> {
```